outputs.tf

```
@@ -37,3 +37,8 @@ output "eks_node_group_cbd_pet_name" {
   description = "The pet name of this node group, if this module generated one"
   value       = join("", random_pet.cbd.*.id)
 }
+
+output "eks_worker_launch_template_name" {
+  description = "The launch template name provided or created for the worker nodes"
+  value       = local.fetch_launch_template ? data.aws_launch_template.this.name : aws_launch_template.default.name
```

```suggestion
  value       = local.enabled ? (local.fetch_launch_template ? data.aws_launch_template.this[0].name : aws_launch_template.default[0].name) : ""
```

Since all those resources have `count`, they are already lists (not single items), to to get an output from them we need to use the list notation `[0]`